### DISCUSSION TITLE
How to combine glyphs from two Type1 fonts in a single Type1 font?

### DISCUSSION THREAD

**Discussion by sl2c** [fonttools/fonttools#3239] [Q&A]
So, I want to create a Type1 (PFB) font with a given encoding, and glyph names in the encoding are such that some of the glyphs are in one existing PFB file, and the others — in another. So, I can get the glyphSet()-s from both of these existing PFB files and pull the glyphs themselves. What should be done next? Can I bluntly write into self["CharStrings"]? If so, should any "rehash" function be called after such editing?

On another note, when trying to save the resulting PFB file I run into problems which have nothing to do with the editing per se. Just filed a suspected bug report on this.

**Comment by justvanrossum** [comment]
Even after the fixes of #3240, this may not be trivial: the font you posted in #3238 contains `/Subrs` (subroutines), and the glyphs (CharStrings) depend on them. If the Subrs in both fonts are not the same, you can't naively merge the charstrings. You may be better off doing this in a font editor.

But again: please reconsider whether the Type 1 font format is really your only choice. It is so extremely obsolete, that even its inventor (Adobe) doesn't support it anymore.

**Comment by sl2c** [comment]
Looks like what I need to do can be done with the following fontforge script:
```
#!/usr/bin/env fontforge
Open($1)
MergeFonts($2)
Generate($2:r + '-out.pfb')
```

**Reply by khaledhosny** [reply]
FontForge supports Python scripting as well, and you can call FontForge Python module from Python scripts, and it should be more than enough for your needs.
https://fontforge.org/docs/scripting/python.html
